code/modules/reagents/reagent_containers/hypospray.dm

```diff
@@ -104,19 +104,26 @@
 /obj/item/reagent_containers/hypospray/autoinjector
 	name = "autoinjector"
 	desc = "A rapid and safe way to administer small amounts of drugs by untrained or trained personnel."
-	icon_state = "autoinjector1"
-	item_state = "autoinjector1"
-	var/empty_state = "autoinjector0"
-	flags = OPENCONTAINER
+	desc_fluff = "Funded by the Stellar Corporate Conglomerate, produced by Zeng-Hu Pharmaceuticals, this autoinjector system was rebuilt from the ground up from the old variant to provide maximum user feedback."
+	desc_info = "Autoinjectors are spent after using them. To re-use, use a screwdriver to open the back panel, then simply pour any desired reagent inside. Use in-hand, or click it while it's in your active hand to prepare it for reuse."
+	icon_state = "autoinjector"
+	item_state = "autoinjector"
+	var/name_label
+	var/spent = TRUE
 	amount_per_transfer_from_this = 5
 	possible_transfer_amounts = null
 	volume = 5
 	time = 0
 
 /obj/item/reagent_containers/hypospray/autoinjector/Initialize()
-	. =..()
-	icon_state = empty_state
-	item_state = empty_state
+	. = ..()
+	if(name_label)
+		name_unlabel = name
```

Comment:
this is just initial(name)

Reply:
this is setting the name for the remove label proc